core/src/main/scala/akka/kafka/javadsl/DiscoverySupport.scala

```diff
@@ -28,24 +29,43 @@ object DiscoverySupport {
    */
   def consumerBootstrapServers[K, V](
       config: Config,
-      system: ActorSystem
+      system: ClassicActorSystemProvider
   ): java.util.function.Function[ConsumerSettings[K, V], CompletionStage[ConsumerSettings[K, V]]] = {
+    implicit val sys: ClassicActorSystemProvider = system
     val function: ConsumerSettings[K, V] => Future[ConsumerSettings[K, V]] =
-      scaladsl.DiscoverySupport.consumerBootstrapServers(config)(system)
+      scaladsl.DiscoverySupport.consumerBootstrapServers(config)
     function.andThen(FutureConverters.toJava).asJava
   }
 
+  @InternalApi // kept for bin-compatibility
```

Comment:
I'm confused why you marked this and the other overloads in `DiscoverySupport` as an internal API.  Shouldn't it be deprecated instead?

Reply:
Internal API might be wrong, you're right.
I couldn't decide if we'd go all in and do `@deprecated` + `private[kafka]`.
Looking at it again, there is no way to call this method when recompiling so we can go all in.

Reply:
Mima dislikes `private[kafka]` nowadays, so I stayed away from it.